torch/_subclasses/fake_tensor.py

```diff
@@ -267,20 +267,26 @@ def from_real_tensor(
         if type(t) is torch.nn.Parameter:
             assert not make_constant
 
-        def mk_fake_tensor(make_meta_t):
+        maybe_constant_t = t if make_constant else None
+
+        def mk_fake_tensor(make_meta_t, *, orig_t=None):
             # NB: don't use in_kernel_invocation_manager. to
             # ensure FakeTensor can internally do constant computation
             # as necessary.  Invocation manager is "more correct" as
             # it works for more operators in make_meta_t, but
             # invariant is that make_meta_t only calls factories
             # for which it is not strictly necessary to use the
             # invocation manager (I think!)
+            if orig_t is not None:
+                maybe_memo = self._get_memo(orig_t)
+                if maybe_memo is not None:
+                    return maybe_memo
```

Comment:
This is necessary so that if we already fakify the tensor on the nested int, we don't fakify the tensor a second time when we fakify the inner tensors during subclass fakification. Previously the fake tensor creation callback used in subclass fakification did not actually check the FakeTensor memo  (I would've expected it to, but I guess we just never needed to do that, and now that I think about this would've mean that if we had two NestedTenosr as input, their inner offsets wouldn't share FakeTensors? - I haven't checked)